node/src/accountant/scanners.rs

```diff
@@ -1132,80 +1248,217 @@ mod tests {
+         Failed process to be screened for persisted data. Caused by: Blockchain error: Batch processing: \"Attempt failed\". \
+         With signed transactions, each registered: 0x00000000000000000000000000000000000000000000000000000000000015b3, \
+         0x0000000000000000000000000000000000000000000000000000000000003039."));
+        log_handler.exists_log_containing(
+            &format!("WARN: {test_name}: \
+            Deleting fingerprints for failed transactions 0x00000000000000000000000000000000000000000000000000000000000015b3, \
+            0x0000000000000000000000000000000000000000000000000000000000003039",
+        ));
+        //we haven't supplied any result for mark_pending_payable() and so it's proved uncalled
+    }
+
+    #[test]
+    fn payable_scanner_handles_error_born_too_early_to_see_transaction_hash() {
+        init_test_logging();
+        let test_name = "payable_scanner_handles_error_born_too_early_to_see_transaction_hash";
+        let sent_payable = SentPayables {
+            payment_procedure_result: Err(BlockchainError::PayableTransactionFailed {
+                msg: "Some error".to_string(),
+                signed_and_saved_txs_opt: None,
+            }),
+            response_skeleton_opt: None,
+        };
+        let mut subject = PayableScannerBuilder::new().build();
+
+        subject.finish_scan(sent_payable, &Logger::new(test_name));
+
+        let log_handler = TestLogHandler::new();
+        log_handler.exists_log_containing(&format!(
+            "DEBUG: {test_name}: Got 0 properly sent payables of not \
+         determinable number of attempts"
+        ));
+        log_handler.exists_log_containing(&format!(
+            "DEBUG: {test_name}: Ignoring a non-fatal error on our end from before \
+            the transactions are hashed: LocallyCausedError(PayableTransactionFailed \
+             {{ msg: \"Some error\", signed_and_saved_txs_opt: None }})"
+        ));
+    }
+
     #[test]
     #[should_panic(
-        expected = "Payable fingerprint for 0x0000…0315 doesn't exist but should by now; system unreliable"
+        expected = "Expected pending payable fingerprints for (tx: 0x0000000000000000000000000000000000000000000000000000000000000315, \
+     to wallet: 0x000000000000000000000000000000626f6f6761), (tx: 0x0000000000000000000000000000000000000000000000000000000000000315, \
+     to wallet: 0x00000000000000000000000000000061676f6f62) were not found; system unreliable"
     )]
     fn payable_scanner_panics_when_fingerprint_is_not_found() {
-        let now = SystemTime::now();
-        let payment_hash = H256::from_uint(&U256::from(789));
-        let payable = Payable::new(make_wallet("booga"), 6789, payment_hash, now);
-        let pending_payable_dao = PendingPayableDaoMock::default().fingerprint_rowid_result(None);
+        let hash_1 = make_tx_hash(789);
+        let payment_1 = PendingPayable::new(make_wallet("booga"), hash_1);
+        let hash_2 = make_tx_hash(789);
```

Comment:
This hash should be different so that the test can tell if they accidentally get swapped somewhere.